hydra/test_utils/test_utils.py

```diff
@@ -129,10 +133,15 @@ def __call__(self, cfg: DictConfig) -> Any:
         return 100
 
     def __enter__(self) -> "SweepTaskFunction":
-        self.temp_dir = tempfile.mkdtemp()
+
         overrides = copy.deepcopy(self.overrides)
         assert overrides is not None
-        overrides.append(f"hydra.sweep.dir={self.temp_dir}")
+        if self.sweep_dir:
+            self.temp_dir = self.sweep_dir
+        else:
+            self.temp_dir = tempfile.mkdtemp()
+            overrides.append(f"hydra.sweep.dir={self.temp_dir}")
```

Comment:
There has to be a better way to do what you want.

Explain the problem you are solving again?

Reply:
```temp_dir``` is local to the  OS of the machine running the tests, a temp dir on MAC OS looks like ```/var/folders/n_/9qzct77j68j6n9lh0lw3vjqcn96zxl/T/tmpz_w5rxac```, when running on remote ubuntu AWS instances, Hydra will try to create a dir looks like this, which won't work. So I want to manually pass in a ```sweep_dir``` in this case.

Reply:
okay.
should we push down the creation of the working directory for the job itself? It should be creating it if it's not there anyway.

If the actual directory is not there it will try to create it. we just need to choose a directory that can work remotely. a safe bet is `${env:HOME}/tmp/${now:...}`.

I will take a closer look tomorrow, but I think the integration test is taking a tempdir from the caller. can you caller override that tmpdir through the overrides somehow instead of opening a whole new channel through everything to pass it?

Reply:
I thought I responded to this one, but somehow it got lost 🤔

There's 2 test suites:  ```LauncherTestSuite``` and the ```IntegrationTestSuite```. I've fixed the ```IntegrationTestSuite``` with the tmpdir. Pushing the dir creation to the the jobs is a great ideas. I will make the changes now.